src/Voron/Util/NativeList.cs

```diff
@@ -65,6 +65,15 @@ public void Add(ByteStringContext ctx, T value)
         AddUnsafe(value);
     }
 
+    public void Fill(T value)
+    {
+        if (Capacity == 0) 
+            return;
+        
+        Count = Capacity;
```

Comment:
I don't think this is the correct semantic. `.Fill()` should only account for `.Count`. This is more like `.InitializeWithValue()`

Reply:
Agree, there is also the consideration that you use that to init the _rest_ of the buffer, no?
If I have existing values, they are overwritten, no?